### PR TITLE
[#4721] E2E puppeteer

### LINKED ISSUE #4721: End-to-end testing

### PR DESCRIPTION
**Pull request by zuhdil**
- [ ] Test plan | Unit test | Integration test
- [ ] Documentation

Closes #4721

### CODE REVIEW

**Comment by zuhdil** [comment]
To run the test on your local dev environment *(it requires `node` and `npm` to install your environment)*

```sh
./scripts/devhelpers/run-e2e-local.sh
```

If you want to spawn up the browser when running the test add `--spawn` argument

```sh
./scripts/devhelpers/run-e2e-local.sh --spawn
```

**Review by ifirmawan — APPROVED** [review]

File: .branch_name [hunk 1]
```diff
@@ -0,0 +1 @@
+master
```
Comment (MichaelAkvo) [inline .branch_name:1]: Was this committed by accident?

**Review by MichaelAkvo — CHANGES_REQUESTED** [review]

**Review by MichaelAkvo — CHANGES_REQUESTED** [review]
All this really needs is documentation.

File: Dockerfile-e2e [hunk 2]
```diff
@@ -0,0 +1,30 @@
+FROM node:12-slim
+
+RUN apt-get update \
+    && apt-get install -y wget gnupg \
+    && wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - \
+    && sh -c 'echo "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list' \
+    && apt-get update \
+    && apt-get install -y google-chrome-stable fonts-ipafont-gothic fonts-wqy-zenhei fonts-thai-tlwg fonts-kacst fonts-freefont-ttf libxss1 \
+      --no-install-recommends \
+    && rm -rf /var/lib/apt/lists/*
```
[the remark this answers is not in the document]
Comment (zuhdil) [reply Dockerfile-e2e:10]: I took the Dockerfile from [documentation](https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md#running-puppeteer-in-docker) then I remove some comments and change the docker user. It said that it needs to install the necessary libs to make the bundled version of Chromium that Puppeteer. I think it is running a Chromium under the hood.

File: ci/promote-test-to-prod.sh [hunk 3]
```diff
@@ -1,6 +1,14 @@
 #!/usr/bin/env bash
+set -eu
 
 git fetch
 
+echo "Running E2E tests"
+if [[ $(uname -s) == "Linux" ]] && [[ ! "${HEADLESS:-}" = false ]]; then
```
[the remark this answers is not in the document]
Comment (zuhdil) [reply ci/promote-test-to-prod.sh:7]: `HEADLESS=false` is a flag to disable the headless mode, I don't think it's correct semantically to use a value other than `false`

**Review by MichaelAkvo — COMMENTED** [review]
Just one small typo and the rest looks good to me. Also tested it on my computer (the "linux" script) and it worked.

**Review by MichaelAkvo — APPROVED** [review]
LGTM

**Pull request merged**